x-pack/legacy/plugins/upgrade_assistant/server/np_ready/types.ts

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { Legacy } from 'kibana';
+import { SavedObjectsClientContract } from 'kibana/server';
+import { ServerRoute } from 'hapi';
+import { ElasticsearchPlugin } from 'src/legacy/core_plugins/elasticsearch';
+import { XPackMainPlugin } from '../../../xpack_main/xpack_main';
+
+export type RegisterRoute = (args: ServerRoute) => void;
+
+export interface ServerShim {
+  usage: {
+    collectorSet: {
+      makeUsageCollector: any;
+      register: any;
+    };
+  };
+  plugins: {
+    elasticsearch: ElasticsearchPlugin;
+    xpack_main: XPackMainPlugin;
+  };
+  route: RegisterRoute;
+  log: any;
+  events: any;
+  savedObjects: Legacy.SavedObjectsService;
```

Comment:
This is quite a big dependency, so it might be useful to break it up to make it clearer what exactly you depend on, I think you only need `savedObjects: {getSavedObjectsRepository: Legacy.SavedObjectsService['getSavedObjectsRepository']}` which will be available once https://github.com/elastic/kibana/pull/48882 merges.